### ISSUE TITLE
Change way how Get-DbaDbFile and Get-DbaDbFileMapping get data

### ISSUE THREAD

**Issue by Zhunya** [sqlcollaborative/dbatools#7968]
### Summarize Functionality

When the database status is not `Normal`, let's query the system view `master.sys.master_files` instead of reporting an error.

```powershell
WARNING: [08:52:52][Get-DbaDbFile] Failure | Database '<databaseName>' cannot be opened. It is in the middle of a restore.
```

### Is there a command that is similiar or close to what you are looking for?

No

**Comment by wsmelton** [comment]
Querying `sys.master_files` would require more permissions than the command requires currently.

**Comment by Zhunya** [comment]
We are database administrators, we have all the necessary rights 💯

**Comment by wsmelton** [comment]
You cannot make an assumption that (1) every user of our module is a DBA nor (2) has the explicit permissions to perform their job in any given environment.

The command as-is if we assume SMO only queries `sys.database_files` only requires public role to the database. Adding a query to `sys.master_files` will require one of the following explicit permissions on **all databases**:

- CREATE DATABASE
- ALTER ANY DATABASE
- VIEW ANY DEFINITION

If the user is missing permission on one database for any of the above the query, as a whole, will fail to execute.

**Comment by Zhunya** [comment]
Yes, sorry, wrong assumptions. Then check if the user has the necessary rights, in case the database is inaccessible?

**Comment by wsmelton** [comment]
Updated, worded wrong as those are instance level permissions.

**Comment by Zhunya** [comment]
The proposed change to `Get-DbaDbFile` may be very useful in a scenario where the database is restored with some randomness in file names:

- 04:00 - *First Job* - FULL `DB_COPY` Restore
  - We have dynamic file names supplied here with help of `DestinationFileSuffix` and current date and time - `$destinationFileSuffix = "_NIGHT_$(Get-Date -Format 'yyyyMMddHHmm')"`
  - NoRecovery - $True
- 08:00 - *Second Job* DIFF backup was taken, restore it to `DB_COPY`
  - In this moment we cant get file names to supply to `Restore-DbaDatabase` with the `FileMapping` parameter
At the last step Restore-DbaDatabase just did some work and did non restore DIFF.

Rare situation due to random database file names.

**Comment by potatoqualitee** [comment]
I traced the current implementation and the fallback is technically possible, but it is not a drop-in query replacement.

### What works

When a database is `RESTORING`, `RECOVERY_PENDING`, `SUSPECT`, or `OFFLINE`, querying that database's `sys.database_files` is unavailable or unreliable. Microsoft explicitly documents `sys.master_files` as the place to inspect file state for an offline database: https://learn.microsoft.com/en-us/sql/relational-databases/databases/file-states?view=sql-server-ver17

A fallback query in `master` can return the useful core needed by restore/file-mapping workflows:

- database ID/name
- logical and physical file names
- type/type description
- size, max size, growth, percent-growth flag
- file state
- data-space ID

That would cover the issue's main scenario: obtaining logical/physical names between a full `NORECOVERY` restore and a later differential restore.

### Drawbacks and compatibility risks

1. **Higher server permissions.** Microsoft documents the minimum visibility for a row in `sys.master_files` as `CREATE DATABASE`, `ALTER ANY DATABASE`, or `VIEW ANY DEFINITION`: https://learn.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-master-files-transact-sql?view=sql-server-ver17. The current database-context query can work for callers with narrower database access. A transparent fallback would therefore work for DBAs but still fail for some least-privilege callers.

2. **The existing output is richer than `sys.master_files`.** `Get-DbaDbFile` currently joins `sys.database_files`, `sys.filegroups`, `sys.dm_io_virtual_file_stats`, and `sys.dm_os_volume_stats`. From `master`, we cannot faithfully populate filegroup name/default/read-only metadata, used space, I/O counters, or volume free space for an inaccessible database. Returning zero or null for those fields must be deliberate and documented; silently fabricating them would be worse than the current warning.

3. **Mixed output paths.** `Get-DbaDbFile` currently processes SMO Database input and executes per-database queries. A fallback object must keep the same property names and types (especially `dbasize`) or downstream consumers such as `Get-DbaDbFileMapping` can break.

4. **State-specific semantics.** `sys.master_files` is authoritative for configured file metadata, but not all runtime values. For example, Microsoft notes that its tempdb sizes are startup templates, not current growth. The fallback should be restricted to inaccessible user databases, not become the default query.

5. **Database visibility.** An explicitly requested inaccessible database should produce a clear permission error if its row cannot be seen. It should not be indistinguishable from a nonexistent database.

### Recommended design

Keep the current database-context query for accessible databases. For an explicitly selected inaccessible database, fall back to a single parameterized query against `master.sys.master_files`. Return the existing output shape, populate only values that are authoritative, and set unavailable runtime/filegroup properties to `$null` (not zero). Emit a verbose message identifying the fallback and a targeted warning if the caller lacks the required permission.

I would also make `Get-DbaDbFileMapping` consume this fallback naturally rather than add its own second implementation. Tests should cover an inaccessible mocked database, output type/shape parity, permission failure, and confirmation that accessible databases stay on the current path.

This is implementable, but the permission and partial-metadata contract should be accepted before coding because it is the public behavior change.
